java-lib/src/test/java/org/clink/feature/OneHotEncoderTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.clink.feature;
+
+import org.apache.flink.ml.feature.onehotencoder.OneHotEncoder;
+import org.apache.flink.ml.feature.onehotencoder.OneHotEncoderModel;
+import org.apache.flink.ml.linalg.Vectors;
+import org.apache.flink.ml.util.ReadWriteUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+
+import org.clink.feature.onehotencoder.ClinkOneHotEncoderModel;
+import org.clink.feature.onehotencoder.ClinkOneHotEncoderUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.Assert.assertEquals;
+
+/** Tests Java wrapped C++ OneHotEncoderModel Operator. */
+public class OneHotEncoderTest {
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Test
+    public void testFitAndPredict() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.createLocalEnvironment();
+        env.setParallelism(1);
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+        OneHotEncoder estimator = new OneHotEncoder();
+        estimator.setInputCols("a", "b");
+        estimator.setOutputCols("c", "d");
+        estimator.setDropLast(false);
+
+        DataStream<Row> trainStream = env.fromElements(Row.of(0, 1), Row.of(2, 3));
+        Table trainTable = tEnv.fromDataStream(trainStream).as("a", "b");
+        DataStream<Row> predictStream = env.fromElements(Row.of(0, 1));
+        Table predictTable = tEnv.fromDataStream(predictStream).as("a", "b");
+
+        OneHotEncoderModel model = estimator.fit(trainTable);
+
+        Path saveDir = Files.createTempDirectory("temp");
+        String savePath = saveDir.toAbsolutePath().toString();
+        ReadWriteUtils.saveMetadata(model, savePath);
+        ClinkOneHotEncoderUtils.saveModelDataAsProtobuf(model, savePath);
```

Comment:
Hmm.. do we expect every algorithm to have a `XXXUtils` class and ask users to manually invoke `XXXUtils::saveModelDataAsProtobuf`?

I thought the plan is to implement `ClinkOneHotEncoder` (which implements `Estimator`) and `ClinkOneHotEncoderModel` so that users don't have to know/do the extra conversion?

Reply:
OK. I'll make the change in this way.